File: backend/app/auth/schemas.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ErrorBody(BaseModel):
    code: str
    message: str


class ErrorResponse(BaseModel):
    success: bool = False
    error: ErrorBody


class AuthenticatedUser(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

    id: str
    email: str
    full_name: str = Field(serialization_alias="fullName")
    avatar_url: str | None = Field(default=None, serialization_alias="avatarUrl")
    provider: str
    created_at: datetime = Field(serialization_alias="createdAt")
    updated_at: datetime = Field(serialization_alias="updatedAt")


class AuthenticatedUserResponse(BaseModel):
    success: bool = True
    data: AuthenticatedUser


class LogoutResponse(BaseModel):
    success: bool = True
    message: str

File: frontend/src/features/auth/schemas.ts
import { z } from 'zod'

export const authenticatedUserSchema = z.object({
  id: z.string(),
  email: z.string().email(),
  fullName: z.string(),
  avatarUrl: z.string().nullable(),
  provider: z.string(),
  createdAt: z.string(),
  updatedAt: z.string(),
})

export const authenticatedUserResponseSchema = z.object({
  success: z.boolean(),
  data: authenticatedUserSchema,
})

export const logoutResponseSchema = z.object({
  success: z.boolean(),
  message: z.string(),
})

export type AuthenticatedUser = z.infer<typeof authenticatedUserSchema>
